#
# Committed-By: Marc Kleine-Budde <mkl@pengutronix.de>, 2006-11-29
#
# Error:
#
# check for MD5 is not cross aware, make it cacheable and added cross default value
#
# State:
#
# should be commited
#
Index: pure-ftpd-1.0.21/configure.ac
===================================================================
--- pure-ftpd-1.0.21.orig/configure.ac
+++ pure-ftpd-1.0.21/configure.ac
@@ -1046,7 +1046,9 @@ else
 fi
 
 
-AC_MSG_CHECKING(whether you already have a standard MD5 implementation)
+AC_CACHE_CHECK([whether you already have a standard MD5 implementation],
+  [ac_cv_use_system_crypt_md5],
+[
 AC_RUN_IFELSE([AC_LANG_SOURCE([[
 #include <stdio.h>
 #include <stdio.h>
@@ -1066,12 +1068,15 @@ int main(void)
 
     return strcasecmp(b, "098f6bcd4621d373cade4e832627b4f6");
 }
-]])],[
-AC_MSG_RESULT(yes)
-AC_DEFINE(USE_SYSTEM_CRYPT_MD5,,[Define if you already have standard
+]])],[ac_cv_use_system_crypt_md5=yes
+],[ac_cv_use_system_crypt_md5=no
+],[ac_cv_use_system_crypt_md5="no (cross compiling)"])
+])
+if test "x$ac_cv_use_system_crypt_md5" = xyes
+then
+  AC_DEFINE(USE_SYSTEM_CRYPT_MD5,,[Define if you already have standard
 MD5 functions])
-],[AC_MSG_RESULT(no)
-],[])
+fi
 
 AC_MSG_CHECKING(whether you already have a standard SHA1 implementation)
 AC_RUN_IFELSE([AC_LANG_SOURCE([[
